Model/Config/Source/Available/Available.php

```diff
@@ -157,20 +163,23 @@ protected function configureSDK()
         return false;
     }
 
-    protected function getPaymentOptionId()
+    protected function setCurrentMethode()
     {
-        $method = $this->_paymentmethodFactory->create($this->_class);
-        if ($method instanceof PaymentMethod) {
-            return $method->getPaymentOptionId();
+        $this->currentMethod = $this->_paymentmethodFactory->create($this->_class);        
```

Comment:
setCurrentMethode renamen naar getCurrentMethod. 
Deze returned this->currentMethod.
De create alleen aanroepen als currentmethod emty is. 

De getpaymentoptionId en getpaymentoptioncode  roepen getCurrentMethod aan. 

Bij de _isAvailable functie weghalen (wordt dus geset zodra er naar gevraagd word).